freqtrade/main.py

```diff
@@ -400,15 +400,42 @@ def create_trade(stake_amount: float, interval: int) -> bool:
     else:
         return False
 
-    # Calculate amount
+    min_qty = None
+    max_qty = None
+    step_qty = None
+
+    (min_qty, max_qty, step_qty) = exchange.get_trade_qty(pair)
+
+    # Calculate bid price
     buy_limit = get_target_bid(exchange.get_ticker(pair))
+
+    # Calculate base amount
     amount = stake_amount / buy_limit
 
-    order_id = exchange.buy(pair, buy_limit, amount)
+    # if amount above max qty: just buy max qty
+    if max_qty:
```

Comment:
It looks like this behavior belong to Binance only. I think main.py should be decouple of Exchanges behavior. Can this logic be part of the Binance class (buy() method)? (Line 416 to 434)

Reply:
the thing is, you also need the correct trade quantity information to be able to send the correct Telegram information and store the correct stake in sqlite; because the stake of the trade changes. So your desired stake might be 0.05 BTC, but due to order limits the trade will change to 0.04999 BTC etc

Reply:
also need this logic in main.py to be able to handle timed out partially filled sell orders now. Please re-review